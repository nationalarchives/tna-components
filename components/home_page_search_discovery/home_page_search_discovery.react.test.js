import React from 'react';
import TestRenderer from 'react-test-renderer';
import HomePageSearchDiscovery from './home_page_search_discovery.react';
import Form from '../global/form/form.react';
import Input from '../global/form/input.react';
import Select from '../global/form/select.react';

let testRenderer, testInstance;

beforeAll(() => {
  testRenderer = TestRenderer.create(<HomePageSearchDiscovery />);
  testInstance = testRenderer.root;
});

describe('HomePageSearchDiscovery component', () => {
  it('=> Form Props are the correct ones', () => {
    expect(testInstance.findByType(Form).props.method).toBe('get');
    expect(testInstance.findByType(Form).props.action).toBe(
      'https://discovery.nationalarchives.gov.uk/results/r'
    );
  });
  it('=> Form elements exists', () => {
    expect(testInstance.findAllByType(Input).length).toEqual(4);
    expect(testInstance.findAllByType(Select).length).toEqual(1);
  });
  it('=> onChangeInput', () => {
    const event = (id, value) => {
      return {
        preventDefault() {},
        target: {
          id,
          value
        }
      };
    };

    expect(testInstance.findAllByType(Input)[0].props.value).toBe('');
    expect(testInstance.findAllByType(Input)[1].props.value).toBe('');
    expect(testInstance.findAllByType(Input)[2].props.value).toBe('');

    testInstance
      .findAllByType(Input)[0]
      .props.onChange(event('search-all-collections', 'show input'));

    expect(testInstance.findAllByType(Input)[0].props.value).toBe('show input');
  });

  it('DOM snapshot test', () => {
    let tree = testRenderer.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
