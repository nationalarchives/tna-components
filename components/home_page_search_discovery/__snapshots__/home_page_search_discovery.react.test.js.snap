// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageSearchDiscovery component DOM snapshot test 1`] = `
<div
  id="discovery-homepage-search-react-component"
>
  <form
    action="/results/r"
    id="discoveryHomePageSearch"
    method="get"
    name="discoveryHomePageSearch"
    onSubmit={[Function]}
  >
    <div
      className="grid-container"
      role="search"
    >
      <div
        className="Headline"
      >
        <h1>
          Explore The National Archives and more than 2,500 other archives
        </h1>
      </div>
      <div
        className="MainSearch"
      >
        <label>
          Search for
        </label>
        <input
          className={null}
          id="search-all-collections"
          name="_q"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Enter keyword(s)"
          type="text"
        />
      </div>
      <div
        className="InputBetween"
      >
        <label>
          Between
        </label>
        <input
          className={null}
          id="start-date"
          name="_sd"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="Start year"
          type="text"
        />
      </div>
      <div
        className="InputAnd"
      >
        <label>
          and
        </label>
        <input
          className={null}
          id="end-date"
          name="_ed"
          onChange={[Function]}
          onKeyUp={[Function]}
          placeholder="End year"
          type="text"
        />
      </div>
      <div
        className="InputAcross"
      >
        <label>
          Held by
        </label>
        <select
          id="held-by"
          name="_hb"
          onChange={[Function]}
          value="all"
        >
          <option
            value=""
          >
            All archives
          </option>
          <option
            value="tna"
          >
            The National Archives only
          </option>
          <option
            value="oth"
          >
            Other archives only
          </option>
        </select>
      </div>
      <div
        className="MoreOptions"
      >
        <a
          href="https://discovery.nationalarchives.gov.uk/advanced-search"
        >
          Advanced search
        </a>
         
        or 
        <a
          href="https://discovery.nationalarchives.gov.uk/browse"
        >
          browse
        </a>
      </div>
      <div
        className="Search"
      >
        <input
          name="_dhs"
          type="hidden"
        />
        <button
          type="submit"
        >
          Search
        </button>
      </div>
    </div>
  </form>
</div>
`;
